### PR TITLE
Added Marshal

### PR DESCRIPTION
**Pull request by bre7**
https://github.com/utahiosmac/Marshal

### CODE REVIEW

File: SerpentComparison.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -7,6 +7,8 @@
 	objects = {
 
 /* Begin PBXBuildFile section */
+		245AA8C61E6EF3E000C91BD7 /* Marshal.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 245AA8C51E6EF3E000C91BD7 /* Marshal.framework */; };
+		245AA8C71E6EF3E000C91BD7 /* Marshal.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 245AA8C51E6EF3E000C91BD7 /* Marshal.framework */; };
```
Comment (dominik-hadl) [inline SerpentComparison.xcodeproj/project.pbxproj]: Somehow the file got added twice. Can you double check it's not added twice in the build phases? :)

Comment (bre7) [reply SerpentComparison.xcodeproj/project.pbxproj]: 👍 Not sure what happened but I re-added the fw

**Review by dominik-hadl — APPROVED** [review]

File: SerpentComparisonTests/SerpentComparisonTests.swift [hunk 2]
```diff
@@ -267,4 +271,30 @@ class SerpentComparisonTests: XCTestCase {
...
+                print(error)
+            }
+        }
+    }
+
+    func testMarshalSmall() {
+        self.measure {
+            do {
+                let smallMarshalDict = try JSONSerialization.jsonObject(with: self.smallData as Data, options: .allowFragments) as! [String: AnyObject]
+                let _ : [PerformanceTestModel] = try smallMarshalDict.value(for: "data")
```
Comment (mariusc) [inline SerpentComparisonTests/SerpentComparisonTests.swift]: This needs to be `[PerformanceTestSmallModel]`

**Review by mariusc — CHANGES_REQUESTED** [review]
Looks good overall. I only had one change request. The results so far seem impressive. Looking forward to merging this, and I'll update the readme afterwards.

[Later Edit]: I just realised I managed to comment on an outdated diff 🤦‍♂️ . However, the issue is still there. Hope it makes sense 😄

File: SerpentComparisonTests/SerpentComparisonTests.swift (hunk 2 above)
Comment (bre7) [reply SerpentComparisonTests/SerpentComparisonTests.swift]: My bad !

**Comment by bre7** [comment]
One more thing, can I also push a new version of setUp() ?

```swift
var largeData: Data!
    var smallData: Data!
    
    override func setUp() {
        super.setUp()

        let bundle = Bundle(for: type(of: self))
        if let path = bundle.url(forResource: "PerformanceTest", withExtension: "json"),
            let data = try? Data(contentsOf: path) {
            largeData = data
        }
        if let path = bundle.url(forResource: "PerformanceSmallTest", withExtension: "json"),
            let data = try? Data(contentsOf: path) {
            smallData = data
        }
    }
```

To avoid NSData -> Data casting inside each method.

File: SerpentComparisonTests/SerpentComparisonTests.swift (hunk 2 above)
Comment (bre7) [reply SerpentComparisonTests/SerpentComparisonTests.swift]: Use XCTFail instead of print ?

**Comment by dominik-hadl** [comment]
@bre7 I agree, makes sense not to cast it everywhere. However, let's make that another PR as it's not related to adding Marshal :)

**Review by mariusc — APPROVED** [review]

**Comment by mariusc** [comment]
Thanks; waiting for Travis to build successfully then we'll merge.

**Comment by bre7** [comment]
I avoided using custom operators, but for the record (and your README), Marshal does provide one `<|`

**Comment by dominik-hadl** [comment]
@bre7 Thanks for the PR, merged it now 🌮 🎉

**Comment by bre7** [comment]
Awesome ! I submitted a new one 🤓

**Pull request merged**